soroban-sdk/src/crypto/bls12_381.rs

```diff
@@ -101,6 +158,61 @@ impl_bytesn_repr!(G2Affine, 192);
 impl_bytesn_repr!(Fp, 48);
 impl_bytesn_repr!(Fp2, 96);
 
+impl Fp {
+    pub fn env(&self) -> &Env {
+        self.0.env()
+    }
+
+    /// Maps to a `G1Affine` point via [simplified SWU
+    /// mapping](https://www.rfc-editor.org/rfc/rfc9380.html#name-simplified-swu-for-ab-0)
```

Comment:
added comment about curve-point mapping to address https://github.com/stellar/rs-soroban-sdk/pull/1449#discussion_r1987321534